### PR TITLE
#12375: Add qid and optional tensor output to ttnn.gelu_bw

### PR DESCRIPTION
**Pull request by mcw-anasuya**
### Ticket
Link to Github Issue: #12375 

### Problem description
Add queue_id and optional output tensors to ttnn.gelu_bw

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10810592006) - PASSED
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/backward/test_backward_gelu.py [hunk 1]
```diff
@@ -55,3 +55,64 @@ def test_bw_gelu_default(input_shapes, device):
...
+    "approximate",
+    (
+        "none",
+        "tanh",
+    ),
+)
+def test_bw_gelu_opt_output(input_shapes, approximate, device):
+    in_data, input_tensor = data_gen_with_range(input_shapes, -100, 100, device, True)
+    grad_data, grad_tensor = data_gen_with_range(input_shapes, -5, 5, device)
+    _, input_grad = data_gen_with_range(input_shapes, -1, 1, device)
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/backward/test_backward_gelu.py:78]: use L1 mem config

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 2]
```diff
@@ -438,7 +438,15 @@ std::vector<Tensor> ExecuteBackwardBiasGelu::invoke(
     TT_FATAL((approximate == "none" || approximate == "tanh") && "Incorrect approximation type (expected 'none', 'tanh')");
     std::vector<Tensor> grad_tensor;
     Tensor input = ttnn::add(input_a, input_b);
-    grad_tensor = ttnn::gelu_bw(grad, input, approximate = approximate, output_mem_config);
+    // grad_tensor = ttnn::gelu_bw(grad, input, approximate = approximate, output_mem_config);
```
Comment (Aswinmcw) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:441]: comment can be removed?

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 3]
```diff
@@ -438,7 +438,15 @@ std::vector<Tensor> ExecuteBackwardBiasGelu::invoke(
...
+    std::vector<std::optional<Tensor>> gelu_result = ttnn::gelu_bw(grad, input, approximate, output_mem_config);
+    for (const auto& opt_tensor : gelu_result) {
+        if (opt_tensor.has_value()) {
+            grad_tensor.push_back(opt_tensor.value());
+        } else {
+            throw std::runtime_error("Gelu backward returned an empty tensor.");
+        }
+    }
```
Comment (Aswinmcw) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:449]: it looks like unary op, why added in binary?

**Review by Aswinmcw — CHANGES_REQUESTED** [review]

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by eyonland — APPROVED** [review]

**Pull request merged**